addons/account/static/src/components/account_move_list/account_move_list.js

```diff
@@ -128,15 +130,34 @@ AccountMoveUploadListRenderer.components = {
     AccountDropZone,
 };
 
-export class AccountMoveUploadListController extends ListController {}
-AccountMoveUploadListController.components = {
+export class AccountMoveListController extends ListController {
+    setup() {
+        super.setup();
+        this.orm = useService("orm");
+    }
+
+    async onDeleteSelectedRecords() {
+        const moveIds = this.model.root.selection.map((r) => r.resId);
+        const isMoveEndOfChain = await this.orm.call('account.move', 'check_move_sequence_chain', [moveIds]);
+        if (!isMoveEndOfChain) {
+            const dialogProps = super.deleteConfirmationDialogProps;
+            const message = this.env._t("This operation will create a gap in the sequence.");
+            dialogProps.body = markup(`<div class="text-danger">${escape(message)}</div>${dialogProps.body}`);
+            this.dialogService.add(ConfirmationDialog, dialogProps);
+        } else {
+            super.onDeleteSelectedRecords(...arguments);
+        }
+    }
```

Comment:
Seems like a lot of copy paste, maybe extract in a mixin?

[Keep it DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)

Reply:
What about using a service?
```diff
diff --git a/addons/account/static/src/components/account_move_form/account_move_form.js b/addons/account/static/src/components/account_move_form/account_move_form.js
index ce1602084848..affb22d41575 100644
--- a/addons/account/static/src/components/account_move_form/account_move_form.js
+++ b/addons/account/static/src/components/account_move_form/account_move_form.js
@@ -8,21 +8,16 @@ import { FormCompiler } from "@web/views/form/form_compiler";
 import { FormRenderer } from "@web/views/form/form_renderer";
 import { FormController } from '@web/views/form/form_controller';
 import { useService } from "@web/core/utils/hooks";
-import { addDeletionDialog } from "../account_move_helpers/account_move_helpers"
 
 export class AccountMoveController extends FormController {
     setup() {
         super.setup();
-        this.orm = useService("orm");
+        this.account_move_service = useService("account_move");
     }
 
     async deleteRecord() {
-        const moveIds = this.model.root.resId;
-        const isMoveEndOfChain = await this.orm.call('account.move', 'check_move_sequence_chain', [moveIds]);
-        if (!isMoveEndOfChain) {
-            addDeletionDialog(super.deleteConfirmationDialogProps, this.env, this.this.dialogService);
-        } else {
-            super.deleteRecord(...arguments);
+        if ( !await this.account_move_service.addDeletionDialog(this, this.model.root.resId) ) {
+            return super.deleteRecord(...arguments);
         }
     };
 };
diff --git a/addons/account/static/src/components/account_move_helpers/account_move_helpers.js b/addons/account/static/src/components/account_move_helpers/account_move_helpers.js
deleted file mode 100644
index e84446ad1977..000000000000
--- a/addons/account/static/src/components/account_move_helpers/account_move_helpers.js
+++ /dev/null
@@ -1,13 +0,0 @@
-/** @odoo-module **/
-
-import { ConfirmationDialog } from "@web/core/confirmation_dialog/confirmation_dialog";
-import { escape } from "@web/core/utils/strings";
-
-import { markup } from "@odoo/owl";
-
-
-export function addDeletionDialog(confirmationDialogProps, env, dialogService) {
-    const message = env._t("This operation will create a gap in the sequence.");
-    confirmationDialogProps.body = markup(`<div class="text-danger">${escape(message)}</div>${confirmationDialogProps.body}`);
-    dialogService.add(ConfirmationDialog, confirmationDialogProps);
-}
diff --git a/addons/account/static/src/components/account_move_list/account_move_list.js b/addons/account/static/src/components/account_move_list/account_move_list.js
index 60460ee96b58..73d037320d34 100644
--- a/addons/account/static/src/components/account_move_list/account_move_list.js
+++ b/addons/account/static/src/components/account_move_list/account_move_list.js
@@ -13,7 +13,6 @@ import { KanbanDropdownMenuWrapper } from "@web/views/kanban/kanban_dropdown_men
 import { KanbanRecord } from "@web/views/kanban/kanban_record";
 import { FileUploader } from "@web/views/fields/file_handler";
 import { standardWidgetProps } from "@web/views/widgets/standard_widget_props";
-import { addDeletionDialog } from "../account_move_helpers/account_move_helpers"
 
 import { Component, useState } from "@odoo/owl";
 
@@ -132,16 +131,12 @@ AccountMoveUploadListRenderer.components = {
 export class AccountMoveListController extends ListController {
     setup() {
         super.setup();
-        this.orm = useService("orm");
+        this.account_move_service = useService("account_move");
     }
 
     async onDeleteSelectedRecords() {
-        const moveIds = this.model.root.selection.map((r) => r.resId);
-        const isMoveEndOfChain = await this.orm.call('account.move', 'check_move_sequence_chain', [moveIds]);
-        if (!isMoveEndOfChain) {
-            addDeletionDialog(super.deleteConfirmationDialogProps, this.env, this.this.dialogService);
-        } else {
-            super.onDeleteSelectedRecords(...arguments);
+        if ( !await this.account_move_service.addDeletionDialog(this, this.model.root.selection.map((r) => r.resId)) ) {
+            return super.onDeleteSelectedRecords(...arguments);
         }
     }
 };
diff --git a/addons/account/static/src/components/account_move_service.js b/addons/account/static/src/components/account_move_service.js
new file mode 100644
index 000000000000..d1fce8a508b6
--- /dev/null
+++ b/addons/account/static/src/account_move_service.js
@@ -0,0 +1,28 @@
+/** @odoo-module **/
+
+import { ConfirmationDialog } from "@web/core/confirmation_dialog/confirmation_dialog";
+import { escape } from "@web/core/utils/strings";
+import { markup } from "@odoo/owl";
+import { registry } from "@web/core/registry";
+
+
+export const accountMove = {
+    dependencies: ["dialog", "orm"],
+    start(env, { dialog, orm }) {
+        return {
+            async addDeletionDialog (component, moveIds) {
+                const isMoveEndOfChain = await orm.call('account.move', 'check_move_sequence_chain', [moveIds]);
+                if ( !isMoveEndOfChain ) {
+                    const message = env._t("This operation will create a gap in the sequence.");
+                    const confirmationDialogProps = component.deleteConfirmationDialogProps;
+                    confirmationDialogProps.body = markup(`<div class="text-danger">${escape(message)}</div>${escape(confirmationDialogProps.body)}`);
+                    dialog.add(ConfirmationDialog, confirmationDialogProps);
+                    return true;
+                }
+                return false;
+            }
+        }
+    }
+}
+
+registry.category("services").add("account_move", accountMove);
```